File: SampleModule/library/src/main/java/com/example/samplemodule/SampleModule.java
package com.example.samplemodule;

import android.content.Context;

import com.artech.externalapi.ExternalApiDefinition;
import com.artech.externalapi.ExternalApiFactory;
import com.artech.framework.GenexusModule;
import com.artech.usercontrols.UcFactory;
import com.artech.usercontrols.UserControlDefinition;

public class SampleModule implements GenexusModule {

	@Override
	public void initialize(Context context) {
		ExternalApiDefinition basicExternalObject = new ExternalApiDefinition(
				BasicExternalObject.class.getSimpleName(),
				BasicExternalObject.class
		);
		ExternalApiFactory.addApi(basicExternalObject);

		UserControlDefinition basicUserControl = new UserControlDefinition(
				BasicUserControl.class.getSimpleName(),
				BasicUserControl.class
		);
		UcFactory.addControl(basicUserControl);
	}
}

File: SampleModule/library/src/main/java/com/example/samplemodule/SampleModule.kt
package com.example.samplemodule

import android.content.Context
import com.genexus.android.core.externalapi.ExternalApiDefinition
import com.genexus.android.core.externalapi.ExternalApiFactory
import com.genexus.android.core.framework.GenexusModule
import com.genexus.android.core.usercontrols.UcFactory
import com.genexus.android.core.usercontrols.UserControlDefinition

class SampleModule : GenexusModule {
	override fun initialize(context: Context) {
		val basicExternalObject = ExternalApiDefinition(
				BasicExternalObject.NAME,
				BasicExternalObject::class.java
		)
		ExternalApiFactory.addApi(basicExternalObject)

		val basicUserControl = UserControlDefinition(
				BasicUserControl.NAME,
				BasicUserControl::class.java
		)
		UcFactory.addControl(basicUserControl)
	}
}